### PR TITLE
Unify committee/da terminology

### LINKED ISSUE #2898: [CLEANUP] - Replace `committee` with `da` in type names
### What is this task and why do we need to work on it?

We want to make the names consistent for types that are related to the DA committee, and `da` seems to make more sense than `committee`.

### What work will need to be done to complete this task?

Replace `committee` with `da` in names for the network, membership, vote, etc.

### Are there any other details to include?

Zulip discussion: https://espresso.zulipchat.com/#narrow/stream/311646-Consensus/topic/Minor.3A.20DA.20vs.2E.20committee.20in.20names/near/430901602.

### What are the acceptance criteria to close this issue?

All DA-related types have `da` in their names.

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by rob-maron**
Closes #2898

### This PR: 

Unifies committee and DA terminology to always be "DA" instead.

### This PR does not: 

Change terminology in any of the election types. This is because "committee" there refers to the members of the _voting_ committee.

### CODE REVIEW

File: crates/example-types/src/node_types.rs [hunk 1]
```diff
@@ -70,24 +70,24 @@ pub type StaticMembership = StaticCommittee<TestTypes>;
 
 impl<TYPES: NodeType> NodeImplementation<TYPES> for PushCdnImpl {
     type QuorumNetwork = PushCdnNetwork<TYPES>;
-    type CommitteeNetwork = PushCdnNetwork<TYPES>;
+    type DANetwork = PushCdnNetwork<TYPES>;
```
Comment (ss-es) [inline crates/example-types/src/node_types.rs:73]: I think we may want to call this `DaNetwork` based on https://rust-lang.github.io/api-guidelines/naming.html (`Uuid` vs `UUID`).

(I personally dislike this convention also, but I think we agreed on it in Zulip)

File: crates/task-impls/src/network.rs [hunk 2]
```diff
@@ -309,8 +307,8 @@ impl<
                     maybe_action = Some(HotShotAction::DAVote);
                     (
                         vote.get_signing_key(),
-                        MessageKind::<TYPES>::from_consensus_message(SequencingMessage::Committee(
-                            CommitteeConsensusMessage::DAVote(vote.clone()),
+                        MessageKind::<TYPES>::from_consensus_message(SequencingMessage::DA(
+                            DAConsensusMessage::DAVote(vote.clone()),
```
Comment (ss-es) [inline crates/task-impls/src/network.rs:311]: I guess this should be `Da` and `DaConsensusMessage`

**Review by ss-es — COMMENTED** [review]
Just some very minor naming nitpicks from recent Zulip discussion, but looks good to me otherwise!

**Review by shenkeyao — APPROVED** [review]
`Da` looks so strange to me 😅 but thanks for making the changes!

**Pull request merged**